tests/integration/utils.go

```diff
@@ -272,6 +273,8 @@ func executeTestCase(
 		performAction(s, i, testCase.Actions[i])
 	}
 
+	resetMatchers(s)
```

Comment:
question: Can you explain to me why we need to do this. The matchers are defined on the state of the test so why is it relevant to reset them once the test is done?